editor/components/observe-typing/index.js

```diff
@@ -120,7 +120,8 @@ class ObserveTyping extends Component {
 
 		// Abort early if already typing, or key press is incurred outside a
 		// text field (e.g. arrow-ing through toolbar buttons).
-		if ( isTyping || ! isTextField( target ) ) {
+		// Ignore typing in a block toolbar
+		if ( isTyping || ! isTextField( target ) || target.closest( '.editor-block-toolbar' ) ) {
```

Comment:
> This still leaves open some possibility for issues with things like link inputs in RichText though.

I think triggering isTyping in that case is the correct behavior.

-- 
For me, All inputs and richtext in block edit functions should trigger the behavior. Side UI shouldn't. All side UI is triggered is shown outside the editor content area (Popovers) aside the toolbar, so the fix made sense.

But I understand that wrapping only the `BlockEdit` component with this might also fix the issue. It would have been a good approach too.